pkg/fleets/controller.go

```diff
@@ -693,3 +701,47 @@ func (c *Controller) filterGameServerSetByActive(fleet *agonesv1.Fleet, list []*
 
 	return active, rest
 }
+
+// mergeCounters adds the contents of c2 into c1.
+func mergeCounters(c1, c2 map[string]agonesv1.AggregatedCounterStatus) map[string]agonesv1.AggregatedCounterStatus {
+	if c1 == nil {
+		c1 = make(map[string]agonesv1.AggregatedCounterStatus)
+	}
+
+	for key, val := range c2 {
+		// If the Counter exists in both maps, aggregate the values.
+		if counter, ok := c1[key]; ok {
+			counter.Capacity += val.Capacity
+			counter.Count += val.Count
+			c1[key] = counter
+		} else {
+			c1[key] = *val.DeepCopy()
+		}
+	}
+
+	return c1
+}
+
+// mergeLists adds the contents of l2 into l1.
+func mergeLists(l1, l2 map[string]agonesv1.AggregatedListStatus) map[string]agonesv1.AggregatedListStatus {
+	if l1 == nil {
+		l1 = make(map[string]agonesv1.AggregatedListStatus)
+	}
+
+	for key, val := range l2 {
+		// If the List exists in both maps, aggregate the values.
+		// TODO: Will this cause issues with install/helm/agones/templates/crds/fleet.yaml because in
```

Comment:
This is part of the Status, not the Spec of the Fleet, so they don't create the Counts and Lists as part of the fleet (they create Counts and Lists as part of a GameServer). The purpose of aggregating the values held in Counts and Lists in each gameserver is to allow the user to autoscale the fleet based on a Counts and List value (we haven't implemented this yet, that's next).

Reply:
Understood. My question was actually about the counters/lists in the Spec of the Fleet, since that's where the 1000 limitation was applied, right? If users don't create the Counts and Lists as part of the fleet, why we included them in the Spec of the Fleet?

Reply:
It's under Status in the fleet.yaml, so I'm thinking that's where I should remove the limit?

Reply:
oh sorry my bad, I thought they are on Spec. Let's keep the limit for now. We can call out that the limitations are 1000 on both game server and fleet level.

Reply:
Thinking about this more I think we need to get rid of the limit (and defaults) at the List/Counter level. I'm leaving in place the "maxProperties" limit of 1000 which limits the total number of Lists/Counters.

Reply:
After our conversation there is no limit on the number of GameServers in a GameServerSet or Fleet, so the theoretical limit at the List/Counter level is infinite if there is no limit in place. We'll need further discussion on what this limit should be. In the meantime I'm removing CRD changes from this PR.